larpandoracontent/LArControlFlow/StitchingCosmicRayMergingTool.cc

```diff
@@ -537,11 +537,14 @@ void StitchingCosmicRayMergingTool::StitchPfos(const MasterAlgorithm *const pAlg
         for (const ParticleFlowObject *const pPfo : pfoList) pfoVector.push_back(pPfo);
         std::sort(pfoVector.begin(), pfoVector.end(), LArPfoHelper::SortByNHits);
 
+        PfoVector reducedPfoVector;
+        const ParticleFlowObject *const selectedPfoToEnlarge(this->ReduceToLongestStitch(pfoVector, reducedPfoVector, pPfoToEnlarge, pfoToLArTPCMap));
+
         LArTPCPair stitchedLArTPCs(nullptr, nullptr);
 
         try
         {
-            this->FindStitchedLArTPCs(pfoVector, pfoToLArTPCMap, stitchedLArTPCs);
+            this->FindStitchedLArTPCs(reducedPfoVector, pfoToLArTPCMap, stitchedLArTPCs);
```

Comment:
Whilst in this area, can you also address a pending TODO here, and raise an exception if either of the LArTPC pointers in the LArTPCPair remains as nullptr, just before these pointers are dereferenced, so just before `x0` is declared, and `tpcBoundaryCenterX` is calculated. Thanks.

Reply:
I added a check whether one of the two objects in the LArTPCPair is still nullptr and throw a status_code_failure exception here. However, given the logic in FindStitchedLArTPCs, I think the information when a status_code_not_found could be used for that purpose, and inside the catch throw the exception in that case. Nevertheless, extra check outside that try-catch doesn't hurt?

Reply:
It's not strictly necessary now (and this is why the code is as it is), but I still always feel a bit uncomfortable seeing something initialised to nullptr, assigned to in another function and then dereferenced. The only exception, which is common throughout our codebase actually, is where list addresses are returned by reference, and there we always just check the returned status code. But we're there trusting SDK functionality, rather than "just" one function in this tool.